core/server/helpers/prev_next.js

```diff
@@ -44,21 +67,20 @@ fetch = function fetch(apiOptions, options, data) {
 module.exports = function prevNext(options) {
     options = options || {};
 
-    var data = createFrame(options.data),
-        apiOptions = {
-            include: options.name === 'prev_post' ? 'previous,previous.author,previous.tags' : 'next,next.author,next.tags'
-        };
+    var data = createFrame(options.data);
 
-    if (!options.fn) {
+    // Guard against incorrect usage of the helpers
+    if (!options.fn || !options.inverse) {
         data.error = i18n.t('warnings.helpers.mustBeCalledAsBlock', {helperName: options.name});
         logging.warn(data.error);
         return Promise.resolve();
     }
 
-    if (isPost(this) && this.status === 'published') {
-        apiOptions.slug = this.slug;
-        return fetch(apiOptions, options, data);
-    } else {
+    // Guard against trying to execute prev/next on previews, pages, or other resources
+    if (!isPost(this) || this.status !== 'published') {
```

Comment:
I think a page check is missing here. 
This was previously handled [here](https://github.com/TryGhost/Ghost/blob/master/core/server/models/post.js#L707).
Or we have to add this to our apiOptions above.

Reply:
@kirrg001 I can certainly add an extra check, but this is already handled by the same logic that makes sure we get posts by default when we do a browse call to the post API

The SQL query that gets used is:

```
{ method: 'select',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: 
   [ 'published',
     false,
     'yui2-javascript-carousel-an-update-about-version-2-8',
     '2010-10-10 17:56:20',
     1 ],
  __knexQueryUid: 'fa63caec-1b84-4bd3-a37f-7f36c4a11dc0',
  sql: 'select "posts".* from "posts" where ("posts"."status" = ? and "posts"."page" = ?) and ("posts"."slug" != ? and "posts"."published_at" <= ?) order by "posts"."published_at" DESC limit ?' }
```
OR, for primary tags:

```
{ method: 'select',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: 
   [ 'published',
     false,
     'starting-out-with-zend-framework',
     '2011-10-16 09:54:50',
     'code',
     0,
     'public',
     1 ],
  __knexQueryUid: '143723bf-fe4d-4e3c-85d8-f4490787b16c',
  sql: 'select "posts".* from "posts" left outer join "posts_tags" on "posts_tags"."post_id" = "posts"."id" left outer join "tags" on "posts_tags"."tag_id" = "tags"."id" where ("posts"."status" = ? and "posts"."page" = ?) and ("posts"."slug" != ? and "posts"."published_at" > ? and ("tags"."slug" = ? and "posts_tags"."sort_order" = ? and "tags"."visibility" = ?)) group by "posts"."id" order by "posts"."published_at" ASC limit ?' }
```

Reply:
But it has a different meaning or?
If you use `next_post` on a static page OR if you query for the next post which is NOT a page.